Slim/Formats/Playlists/CUE.pm

```diff
@@ -482,20 +455,12 @@ sub parse {
 		return {};
 	}
 
-	# calc song ending times from start of next song from end to beginning.
-	my $lastpos = $tracks->{$currtrack}->{'END'};
-
-	# If we can't get $lastpos from the cuesheet, try and read it from the original file.
-	if (!$lastpos && $filename) {
```

Comment:
@oleg-kuh This is where we enter an infinite loop. We must not enter this condition when we're done with a file, or it'll go back to `Slim::Formats->readTags()` which in turn will call `Slim::Formats::FLAC->getTag()`, which will call the CUE sheet parser again. Here's a snippet from a backtrace:

```
[20-11-09 20:07:17.2417] Slim::Formats::Playlists::CUE::parse (103) Backtrace:

   frame 0: Slim::Utils::Log::logBacktrace (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 103)
   frame 1: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 2: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 3: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 4: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 5: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 6: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 7: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 8: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 9: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 10: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 11: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 12: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 13: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 14: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 15: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 16: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 17: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 18: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 19: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 20: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 21: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 22: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 23: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 24: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 25: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 26: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 27: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 28: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 29: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 30: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 31: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 32: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 33: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 34: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 35: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
   frame 36: Slim::Formats::readTags (/Users/mh/SynologyDrive/git/server/Slim/Formats/Playlists/CUE.pm line 463)
   frame 37: Slim::Formats::Playlists::CUE::parse (/Users/mh/SynologyDrive/git/server/Slim/Formats/FLAC.pm line 124)
   frame 38: Slim::Formats::FLAC::getTag (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 191)
   frame 39: (eval) (/Users/mh/SynologyDrive/git/server/Slim/Formats.pm line 181)
```

After only a few seconds there already were thousands of recursions...

Reply:
Re-introducing this check would "fix" the recursion for me. @oleg-kuh - any good reason why you removed it?